plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_snap
+short_description: snapshot management of instances in Proxmox VE cluster
+description:
+  - allows you to create/delete snapshots from instances in Proxmox VE cluster
+options:
+  api_host:
+    description:
+      - the host of the Proxmox VE cluster
+    required: true
+    type: str
+  api_user:
+    description:
+      - the user to authenticate with
+    required: true
+    type: str
+  api_password:
+    description:
+      - the password to authenticate with
+      - you can use PROXMOX_PASSWORD environment variable
+    type: str
+  hostname:
+    description:
+      - the instance name
+    type: str
+  vmid:
+    description:
+      - the instance id
+      - if not set, will be fetched from PromoxAPI based on the hostname
+    type: str
+  validate_certs:
+    description:
+      - enable / disable https certificate verification
+    type: bool
+    default: 'no'
+  state:
+    description:
+     - Indicate desired state of the instance snapshot
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  force:
+    description:
+      - For removal from config file, even if removing disk snapshot fails.
+    default: 'no'
+    type: bool
+  vmstate:
+    description:
+      - Snapshot includes RAM
+    default: 'no'
+    type: bool
+  description:
+    description:
+      - Specify the description for the snapshot. Only used on the configuration web interface.
+      - This is saved as a comment inside the configuration file.
+    type: str
+    version_added: '0.2.0'
+  timeout:
+    description:
+      - timeout for operations
+    default: 30
+    type: int
+  snapname:
+    description:
+      - Name of the snapshot that has to be created
+    default: 'ansible_snap'
+    type: str
+    version_added: '0.2.0'
+
+notes:
+  - Requires proxmoxer and requests modules on host. This modules can be installed with pip.
+requirements: [ "proxmoxer", "python >= 2.7", "requests" ]
+author: Jeffrey van Pelt (@Thulium-Drake)
+'''
+
+EXAMPLES = r'''
+- name: Create new container snapshot
+  proxmox_snap:
```

Comment:
Always use FQCN in examples:
```suggestion
  community.general.proxmox_snap:
```

Reply:
fixed!